### ISSUE TITLE
confd doesn't try to reconnect to etcd cluster

### ISSUE THREAD

**Issue by zytek** [kelseyhightower/confd#247]
Bug: confd doesn't try to reconnect to etcd cluster.

```
2015-04-01T12:07:46Z ip-10-105-20-115 /usr/local/bin/confd[29794]: ERROR 501: All the given peers are not reachable (Tried to connect to each peer twice and failed) [0]
2015-04-01T12:07:46Z ip-10-105-20-115 /usr/local/bin/confd[29794]: ERROR 501: All the given peers are not reachable (Tried to connect to each peer twice and failed) [0]
```

Steps to reproduce.

1. Set up *broken* etcd cluster (in my example I had three etcd nodes that failed to connect to each other due to firewall rules)
2. Start confd with watch=true, reading data from this broken etcd node.
3. confd starts to report error 501 (as seen abobe)
4. Fix etcd cluster (in my case: remove firewall rule that prevented etcd nodes from seeing each other)
5. etcd is now healthy
6. confd still reports error 501

Seems like there is no retrying policy implemented in etcd connector (go-etcd?)

**Comment by zytek** [comment]
This should be easily fixable by providing custom CheckRetry function in go-etcd client struct.

**Comment by kelseyhightower** [comment]
@zytek Would you like to submit a PR for this?

**Comment by zytek** [comment]
@kelseyhightower I'll try after the weekend

**Comment by zytek** [comment]
Well, while testing my patch I've found out that newest confd handles etcd failures properly, at least simple case of starting/stopping etcd localhost node (handling of "connection refused" errors) seems to work just fine. There is already a 2 second wait+retry code here:

https://github.com/kelseyhightower/confd/blob/master/resource/template/processor.go#L94-L99

I now think that my original test procedure was flawed and I failed to observe that. ;-) Closing this for now.
